### PR TITLE
GRF-628 : Fix template translation when selecting a language

### PR DESCRIPTION
**Pull request by jeremyBeaucousinAkeneo**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Category/front/src/feature/pages/CategoryEditPage.tsx [hunk 1]
```diff
@@ -56,7 +56,12 @@ const CategoryEditPage: FC = () => {
 
   // locales
   const uiLocale = userContext.get('uiLocale');
-  const [catalogLocale] = useState(userContext.get('catalogLocale'));
+  const userContextLocale = userContext.get('catalogLocale');
+  const [catalogLocale, setCatalogLocale] = useState(userContextLocale);
+
+  const onChangeLocal = (locale: string) => {
```
Comment (olivierpontier) [inline src/Akeneo/Category/front/src/feature/pages/CategoryEditPage.tsx:62]:
```suggestion
  const handleLocaleChanged = (locale: string) => {
```

Comment (olivierpontier) [reply src/Akeneo/Category/front/src/feature/pages/CategoryEditPage.tsx:62]: or 
```suggestion
  const onChangeLocal**e** = (locale: string) => {
```

**Comment by olivierpontier** [comment]
Can we set a short technical meeting in order to see more clearly how the catalogLocale is managed and shared with children's components? Because we are using a state and the userContext service at several levels.

**Comment by antoinetrouve** [comment]
> Can we set a short technical meeting in order to see more clearly how the catalogLocale is managed and shared with children's components? Because we are using a state and the userContext service at several levels.

Definitively yes. I think this is the same for the channel.

**Review by antoinetrouve — APPROVED** [review]

**Review by BryanS29-akeneo — APPROVED** [review]

**Pull request merged**